hack/arktos-up-scale-out-poc.sh

```diff
@@ -0,0 +1,609 @@
+  [[ -n "${KUBELET_PIDS-}" ]] && sudo kill "${KUBELET_PIDS[@]}" 2>/dev/null
+
+  # Check if the proxy is still running
+  [[ -n "${PROXY_PID-}" ]] && mapfile -t PROXY_PIDS < <(pgrep -P "${PROXY_PID}" ; ps -o pid= -p "${PROXY_PID}")
+  [[ -n "${PROXY_PIDS-}" ]] && sudo kill "${PROXY_PIDS[@]}" 2>/dev/null
+
+  # Check if the scheduler is still running
+  [[ -n "${SCHEDULER_PID-}" ]] && mapfile -t SCHEDULER_PIDS < <(pgrep -P "${SCHEDULER_PID}" ; ps -o pid= -p "${SCHEDULER_PID}")
+  [[ -n "${SCHEDULER_PIDS-}" ]] && sudo kill "${SCHEDULER_PIDS[@]}" 2>/dev/null
+
+  # Check if the etcd is still running
+  [[ -n "${ETCD_PID-}" ]] && kube::etcd::stop
+  if [[ "${PRESERVE_ETCD}" == "false" ]]; then
+    [[ -n "${ETCD_DIR-}" ]] && kube::etcd::clean_etcd_dir
+  fi
+
+  # Delete virtlet metadata and log directory
+  if [[ -e "${VIRTLET_METADATA_DIR}" ]]; then
+        echo "Cleanup runtime metadata folder"
+        rm -f -r "${VIRTLET_METADATA_DIR}"
+  fi
+
+  if [[ -e "${VIRTLET_LOG_DIR}" ]]; then
+       echo "Cleanup runtime log folder"
+       rm -f -r "${VIRTLET_LOG_DIR}"
+  fi
+
+  exit 0
+}
+# Check if all processes are still running. Prints a warning once each time
+# a process dies unexpectedly.
+function healthcheck {
+  if [[ -n "${APISERVER_PID-}" ]] && ! sudo kill -0 "${APISERVER_PID}" 2>/dev/null; then
+    warning_log "API server terminated unexpectedly, see ${APISERVER_LOG}"
+    APISERVER_PID=
+  fi
+
+  if [[ -n "${CTLRMGR_PID-}" ]] && ! sudo kill -0 "${CTLRMGR_PID}" 2>/dev/null; then
+    warning_log "kube-controller-manager terminated unexpectedly, see ${CTLRMGR_LOG}"
+    CTLRMGR_PID=
+  fi
+
+  if [[ -n "${WORKLOAD_CTLRMGR_PID-}" ]] && ! sudo kill -0 "${WORKLOAD_CTLRMGR_PID}" 2>/dev/null; then
+    warning_log "workload-controller-manager terminated unexpectedly, see ${WORKLOAD_CONTROLLER_LOG}"
+    WORKLOAD_CTLRMGR_PID=
+  fi
+
+  if [[ -n "${KUBELET_PID-}" ]] && ! sudo kill -0 "${KUBELET_PID}" 2>/dev/null; then
+    warning_log "kubelet terminated unexpectedly, see ${KUBELET_LOG}"
+    KUBELET_PID=
+  fi
+
+  if [[ -n "${PROXY_PID-}" ]] && ! sudo kill -0 "${PROXY_PID}" 2>/dev/null; then
+    warning_log "kube-proxy terminated unexpectedly, see ${PROXY_LOG}"
+    PROXY_PID=
+  fi
+
+  if [[ -n "${SCHEDULER_PID-}" ]] && ! sudo kill -0 "${SCHEDULER_PID}" 2>/dev/null; then
+    warning_log "scheduler terminated unexpectedly, see ${SCHEDULER_LOG}"
+    SCHEDULER_PID=
+  fi
+
+  if [[ -n "${ETCD_PID-}" ]] && ! sudo kill -0 "${ETCD_PID}" 2>/dev/null; then
+    warning_log "etcd terminated unexpectedly"
+    ETCD_PID=
+  fi
+}
+
+function print_color {
+  message=$1
+  prefix=${2:+$2: } # add colon only if defined
+  color=${3:-1}     # default is red
+  echo -n "$(tput bold)$(tput setaf "${color}")"
+  echo "${prefix}${message}"
+  echo -n "$(tput sgr0)"
+}
+
+function warning_log {
+  print_color "$1" "W$(date "+%m%d %H:%M:%S")]" 1
+}
+
+function start_etcd {
```

Comment:
Also start_etcd has same implementation, not sure whether this can be refactored

Reply:
As stated in "Notes to reviewers" section in the description, refactor would be better for a following PR.

Reply:
I will draft a separate doc on usage

Reply:
What is the benefit of refactor later? I only see allowing others to start using the code before refactor is done. We don't have the pressing need for now.

Reply:
It goes beyond the purpose of this PR and I don't think keeping adding tasks to an existing scope of a PR is the right approach. Implementing a function and then refactor is a common practice to keep dev cycle short and prevent regression. If there's no pressing need for now, I'd suggest to discard this PR and have a dedicated task to merge and refactor the scripts by the scalability team.